clar2wasm/tests/wasm-generation/principal.rs

```diff
@@ -0,0 +1,192 @@
+                    })),
+                ),
+            ])
+            .unwrap()
+            .into(),
+        )
+        .unwrap()
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_principal_construct(
+            version_byte in 0x00..=0xff,
+            hash_bytes in buffer(20)
+        ) {
+            let expected_principal = create_principal_from_strings(
+                &format!("{:02X}", version_byte),
+                &hash_bytes.to_string()[2..],
+                None
+            );
+
+            let expected = match version_byte {
+                // Valid range for version_bytes
+                0x00..=0x1f => {
+                    match version_byte {
+                        // Since tests runs on a Testnet version,
+                        // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
+                        // will return an Ok value.
+                        0x1A | 0x15 => Value::okay(expected_principal),
+                        _ => Ok(create_error_construct(0, Some(expected_principal))),
+                    }
+                },
+                _ => Ok(create_error_construct(1, None)),
+            }.unwrap();
+
+            crosscheck(
+                &format!("(principal-construct? 0x{:02X} {hash_bytes})", version_byte),
+                Ok(Some(expected)),
+            );
+        }
+    }
+
+    proptest! {
+        #![proptest_config(runtime_config())]
+
+        #[test]
+        fn crosscheck_principal_destruct(
+            version_byte in 0x00..=0xff,
+            hash_bytes in buffer(20)
+        ) {
+            let expected_principal = create_principal_from_strings(
+                &format!("{:02X}", version_byte),
+                &hash_bytes.to_string()[2..],
+                None
+            );
+
+            if !(PropValue::from(expected_principal.clone()).to_string().contains("'INVALID_C32_ADD")) {
+                let expected = match version_byte {
+                    // Valid range for version_bytes
+                    0x00..=0x1f => {
+                        match version_byte {
+                            // Since tests runs on a Testnet version,
+                            // version_bytes single_sig (0x1A) || multi_sig (0x15), for Testnet,
+                            // will return an Ok value.
+                            0x1A | 0x15 => Value::okay(
+                                TupleData::from_data(vec![
+                                    ("hash-bytes".into(), hash_bytes),
+                                    ("name".into(), Value::Optional(OptionalData { data: None })),
+                                    (
+                                        "version".into(),
+                                        Value::Sequence(SequenceData::Buffer(BuffData {
+                                            data: vec![version_byte as u8],
+                                        })),
```

Comment:
nit
```suggestion
                                        Value::buff_from_byte(version_byte as u8),
```

Reply:
Don't think this change is really need. Several tests already relies on explicit `TypeSignature`.